tests/check-strings/check-strings

```diff
@@ -0,0 +1,158 @@
+        classes.update(re.findall(class_pat_ui if cur_file.suffix == '.ui' else class_pat, text))
+
+
+# the real checks
+
+caption('.gitmodules')
+
+for p in re.findall(r'\[submodule "([^"]+)"\]\s*$', Path('.gitmodules').read_text(errors='replace'), re.MULTILINE):
+    if not Path(p).is_dir():
+        error('.gitmodules', f'Directory does not exist: {p}')
+
+
+caption('locale')
+
+filenames = set()
+classes_found = set()
+for cur_file in Path('data/locale').glob('*.ts'):
+    tree = ElementTree.parse(str(cur_file))
+    root = tree.getroot()
+    for location in root.findall('./context/message/location'):
+        filenames.add(location.attrib['filename'])
+    for location in root.findall('./context/name'):
+        classes_found.add(location.text)
+for f in sorted(filenames):
+    # The files sometimes are relative to data/local and sometimes to the git tree's root...
+    if not Path(f).is_file() and not Path(f'data/locale/{f}').is_file():
+        error('data/locale', f'Source file does not exist: {f}')
+for c in sorted(classes_found):
+    if is_our_class(c) and '::' not in c and c not in classes:
+        error('data/locale', f'Class does not exist in source code: {c}')
+
+
+caption('themes')
+
+for theme in sorted([d for d in Path('data/themes').iterdir() if d.is_dir()]):
+    stylesheet = theme / 'style.css'
+    idents = {
+        item.value[1]
+        for rule in cssutils.parseString(stylesheet.read_text(encoding='utf-8'))
+        if rule.type == rule.STYLE_RULE
+        for selector in rule.selectorList
+        for item in selector.seq
+        if item.type == 'type-selector' and not item.value[1].startswith('Q')
+    }
+    missing_classes = idents - classes
+    for class_in_sheet in sorted(missing_classes):
+        error(str(stylesheet), f'Class does not exist in source code: {class_in_sheet}')
+
+
+caption('patches (checks only plugins/)')
+
+pat = re.compile(r'/(plugins/\S*)', re.MULTILINE)
+calf = re.compile(r'calf/.*/modules\.')  # these are a bit complicated to fix...
+for cur_file in sorted(Path('.').glob('*/patches/*.patch')):
+    if Path(cur_file).is_file():
+        paths_in_patches = set()
+        for line in pat.findall(cur_file.read_text(errors='replace')):
+            if not calf.search(line):
+                paths_in_patches.add(Path(line))
+        for mpath in sorted(paths_in_patches):
+            # in case of LADSPA SWH effects, check that the XML exists, not the C file
+            # (because the C files are not generated until a build is done)
+            if mpath.parent == Path('plugins/LadspaEffect/swh/ladspa/'):
+                mpath = mpath.with_suffix('.xml')
+            if not mpath.is_file():
+                error(str(cur_file), f'Source file does not exist: {str(mpath)}')
+
+
+caption('debian docs (only one string)')
+
+# Checks for caps.html. This gets relevant when #4027 will be merged
+for line in Path('debian/lmms-common.docs').read_text(errors='replace'):
```

Comment:
This doesn't give you line by line. This gives you the entire file contexts and then go through it character by character.
```suggestion
for line in Path('debian/lmms-common.docs').read_text(errors='replace').splintlines():
```